### PR TITLE
Fix RangeError: Invalid time value

### LINKED ISSUE #363: RangeError: Invalid time value
While reviewing the sforms project, I found the following error in DataTimeAnswer.jsx:
sforms-range-error2
sforms-range-error

### PR DESCRIPTION
**Pull request by palagdan**
Fix #363

### CODE REVIEW

**Comment by blcham** [comment]
@palagdan thanks, i created https://github.com/kbss-cvut/s-forms/issues/365

File: src/components/answer/DateTimeAnswer.jsx [hunk 1]
```diff
@@ -14,9 +14,16 @@ const DateTimeAnswer = (props) => {
 
   useEffect(() => {
     if (props.value) {
-      setDate(new Date(parse(props.value, datePickerFormat, new Date())));
+      const parsedDate = parse(props.value, datePickerFormat, new Date());
```
Comment (blcham) [inline src/components/answer/DateTimeAnswer.jsx:17]: Not sure if we want to fix invalid date

**Comment by blcham** [comment]
Regarding timestamp:
image

See also how it used to work:
https://github.com/kbss-cvut/s-forms/pull/366

**Comment by blcham** [comment]
Regarding timestamp, i can think of two variants:

```
{
      "label": "Timestamp in db",
      "@id": "http://vfn.cz/ontologies/fss-form/timestamp-in-db-q",
      "has-layout-class": ["timestamp"],
      "has_answer": "http://onto.fel.cvut.cz/ontologies/custom/model/instance#instance123456-a",
      "@type": "doc:question"
    },
 ```
 
 or 

```
{
      "label": "Timestamp in db",
      "@id": "http://vfn.cz/ontologies/fss-form/timestamp-in-db-q",
      "has-layout-class": ["datetime"],
      "has_answer": "http://onto.fel.cvut.cz/ontologies/custom/model/instance#instance123456-a",
      "@type": "doc:question"
      "timestamp-format": "epoch-milliseconds"
      "form:has-datatype": {
             "@id": "xsd:integer"
       },
    },
 ```

I also asked chatGPT -- https://chatgpt.com/share/67054eed-155c-8006-9c22-ea64d61cde40, who suggested additional:
`"timestamp-format": "epoch-seconds"  // or "epoch-milliseconds"` which seems a good idea, but we need to have default.

**Comment by blcham** [comment]
@palagdan unless you have better idea I would go with variant **B)**.  It seems to me nice as "datetime" is rather indicating frontend properties of component.

There are two fundamental properties:
```
"has-layout-class": ["datetime"],
"timestamp-format": "epoch-milliseconds",
```

Optional property is:
```
  "form:has-datatype": {
    "@id": "xsd:integer"
  }
```

**Comment by palagdan** [comment]
@blcham 
Could you please check the implementation? The empty answer appears the same because the timestamp uses a datetime layout, but it stores the value in epoch milliseconds.

**Comment by palagdan** [comment]
@blcham 
error-datetime

**Comment by palagdan** [comment]
@blcham 
Could you please help me with that? Maybe you have some suggestions about what it could be...
datetime_bug

**Comment by blcham** [comment]
@palagdan thanks for finding the issue, @ledsoft fixed it in new version, i think it is 0.1.3.

**Comment by palagdan** [comment]
@blcham

I removed DATETIME_NUMBER_FORMAT because it was a poor workaround for the numerical representation of the date. The resolveDateTimeFormat function should not need to handle this, as it deals with layouts, not the internal representation of the date. Now it works as expected:

- If the layout has a timestamp format, it will save the date in milliseconds, allowing for the combination of the layout and this format.
- I fixed the number conversion to the date in the last commit, making it possible to retrieve the date from the number.
- I saved the error state so we can discuss how to display it, or if we should display it at all.
- If there is no value, an error is not thrown, and the input shows the date template instead.

**Comment by blcham** [comment]
Good job!

**Pull request merged**